csv2wiki/__init__.py

```diff
@@ -603,31 +615,33 @@ def __init__(self, config, csv_input, null_as_value, msg_out, dry_run_out):
             # There are two kinds of lines:
             #
             #   1) New section indicator (starts with dots)
-            #   2) Column number (starts with a number)
+            #   2) Text line (starts with a pipe)
+            #   3) Comment line (starts with a pound sign)
+            #      - This is not handled explicitly here because
+            #        it's handled by the python config parser
+            #        before we get the sec_map
             #
             # These regexps help us figure out which kind we've got.
             dot_matcher = re.compile("^(\\.+)\\s*(.*)$")
-            col_matcher = re.compile("^(([0-9]+\\s*)+)\\s*.*$")
+            txt_matcher = re.compile("^\\|\\s*(.*)$")
 
             # Because of the way Python parses ConfigParser syntax,
             # the format we get the sec_map in is one big string,
             # splittable on line breaks into a list of lines.
             for line in sec_map.splitlines():
                 # As usual, I wish Python had Lisp-style 'cond'.
-                m = dot_matcher.match(line)
-                if m:
+                if dot_matcher.search(line):
+                    m = dot_matcher.match(line)
```

Comment:
Minor nit: now we're doing this regexp match twice instead of once: one time for the conditional, and one time to get a match object.

If this was a deliberate readability change, well, okay, I guess match optimization here doesn't make *that* much of a difference.  I just wanted to say that the old-school cycle-counter in me got bruised a little reading this.  I'll be okay.  It'll pass.

Reply:
It's for readability, and for optimization.  The other option is nested if/elses, or doing all the matches up front for a readable if/elif/else.  And search has to be less cycles than match, so we only do match when we really know we can succeed at it.